modules/tinymce/src/themes/silver/main/ts/ui/core/color/Settings.ts

```diff
@@ -44,40 +45,12 @@ const defaultColors = [
 const colorCache = ColorCache(10);
 
 const mapColors = (colorMap: string[]): Menu.ChoiceMenuItemSpec[] => {
-  const colors = [];
-
-  const canvas = document.createElement('canvas');
-  canvas.height = 1;
-  canvas.width = 1;
-  const ctx = canvas.getContext('2d');
-
-  const byteAsHex = (colorByte: number, alphaByte: number) => {
-    const bg = 255;
-    const alpha = (alphaByte / 255);
-    const colorByteWithWhiteBg = Math.round((colorByte * alpha) + (bg * (1 - alpha)));
-    return ('0' + colorByteWithWhiteBg.toString(16)).slice(-2).toUpperCase();
-  };
-
-  const asHexColor = (color: string) => {
-    // backwards compatibility
-    if (/^[0-9A-Fa-f]{6}$/.test(color)) {
-      return '#' + color.toUpperCase();
-    }
-    // all valid colors after this point
-    ctx.clearRect(0, 0, canvas.width, canvas.height);
-    // invalid colors will be shown as white - the first assignment will pass and the second may be ignored
-    ctx.fillStyle = '#FFFFFF'; // lgtm[js/useless-assignment-to-property]
-    ctx.fillStyle = color;
-    ctx.fillRect(0, 0, 1, 1);
-    const rgba = ctx.getImageData(0, 0, 1, 1).data;
-    const r = rgba[0], g = rgba[1], b = rgba[2], a = rgba[3];
-    return '#' + byteAsHex(r, a) + byteAsHex(g, a) + byteAsHex(b, a);
-  };
+  const colors: Menu.ChoiceMenuItemSpec[] = [];
 
   for (let i = 0; i < colorMap.length; i += 2) {
     colors.push({
       text: colorMap[i + 1],
-      value: asHexColor(colorMap[i]),
+      value: '#' + Transformations.anyToHex(colorMap[i]).value,
```

Comment:
Should the `#` be in `anyToHex`? To avoid having it added everywhere?

Reply:
It should really be in the `HexColour` logic, as that's supposed to be what generates the value. However, doing that could potentially break a number of things so I'm pretty wary of making that change.

Reply:
I should also add it seems to be by design that `Hex` only contains hex values, not the `#`. Take for example `fromRgba` that won't add the `#`.

Reply:
Have improved the `HexColour` logic a little to ensure the value is normalized at least, however after talking to others there are cases where you may not want the leading `#` so it's best we leave it out and add it in the contexts where it's required.